opensearch/src/main/java/org/opensearch/sql/opensearch/request/ContinuePageRequest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.opensearch.request;
+
+import static org.opensearch.sql.opensearch.request.OpenSearchScrollRequest.DEFAULT_SCROLL_TIMEOUT;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.search.SearchScrollRequest;
+import org.opensearch.search.builder.SearchSourceBuilder;
+import org.opensearch.sql.opensearch.data.value.OpenSearchExprValueFactory;
+import org.opensearch.sql.opensearch.response.OpenSearchResponse;
+
+/**
+ * Scroll (cursor) request is used to page the search. This request is not configurable and has
+ * no search query. It just handles paging through responses to the initial request.
+ * It is used on second and next pagination (cursor) requests.
+ * First (initial) request is handled by {@link InitialPageRequestBuilder}.
+ */
+@EqualsAndHashCode
+public class ContinuePageRequest implements OpenSearchRequest {
+  final String initialScrollId;
+
+  // ScrollId that OpenSearch returns after search.
+  String responseScrollId;
+
+  @EqualsAndHashCode.Exclude
+  @ToString.Exclude
+  @Getter
+  private final OpenSearchExprValueFactory exprValueFactory;
+
+  @EqualsAndHashCode.Exclude
+  private boolean scrollFinished = false;
+
+  public ContinuePageRequest(String scrollId, OpenSearchExprValueFactory exprValueFactory) {
+    this.initialScrollId = scrollId;
+    this.exprValueFactory = exprValueFactory;
+  }
+
+  @Override
+  public OpenSearchResponse search(Function<SearchRequest, SearchResponse> searchAction,
+                                   Function<SearchScrollRequest, SearchResponse> scrollAction) {
+    SearchResponse openSearchResponse = scrollAction.apply(new SearchScrollRequest(initialScrollId)
+        .scroll(DEFAULT_SCROLL_TIMEOUT));
+
+    // TODO if terminated_early - something went wrong, e.g. no scroll returned.
```

Comment:
Is this `TODO` (and other `TODOs` in the PR) meant to be implemented in Phase 2?

Reply:
Good practice is to leave no TODOs behind.
I'll do my best to resolve them ASAP, but some of them are related to design of existing code and can't be fixed easily.
This very TODO related to a terminated search query - plugin doesn't handle such case for any queries.